pkg/daemon/containernetwork/containernetwork.go

```diff
@@ -359,13 +361,20 @@ func ConfigureContainerNic(containerNicName, hostNicName, nodeIfName string, all
 	return nil
 }
 
-func GenerateContainerVethPair(containerID string) (string, string) {
-	return fmt.Sprintf("%s%s", ContainerHostLinkPrefix, containerID[0:12]), fmt.Sprintf("%s%s", containerID[0:12], ContainerInitLinkSuffix)
+func GenerateContainerVethPair(podNamespace, podName string) (string, string) {
+	// A SHA1 is always 20 bytes long, and so is sufficient for generating the
+	// veth name and mac addr.
+	h := sha1.New()
+	h.Write([]byte(fmt.Sprintf("%s.%s", podNamespace, podName)))
+
+	return fmt.Sprintf("%s%s", ContainerHostLinkPrefix, hex.EncodeToString(h.Sum(nil))[:11]),
```

Comment:
why not still using containerID as suffix here?

Reply:
Because felix cannot recognize the container veth pair name only if the suffix is calculated from pod name and namespace. The reference doc in Chinese is https://segmentfault.com/a/1190000039030174.